includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```diff
@@ -409,6 +417,9 @@ public function get_upe_available_payment_methods() {
 		$available_payment_methods = [];
 
 		foreach ( self::UPE_AVAILABLE_METHODS as $payment_method_class ) {
+			if ( ! $payment_method_class::is_available() ) {
+				continue;
+			}
```

Comment:
It seems like `is_available` is the only static function from `WC_Stripe_UPE_Payment_Method`. I'd say we could convert it to a non-static function similar to `requires_automatic_capture`. By doing so:

```php
foreach ( $this->payment_methods as $payment_method ) {
	if ( ! $payment_method->is_available() ) {
		continue;
	}
	$available_payment_methods[] = $payment_method->get_id();
}
return $available_payment_methods;
```

I think this makes more sense also because there's already a loop through `self::UPE_AVAILABLE_METHODS` and a defined object for each payment method at construct.

Reply:
updated in [0ea817f](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2452/commits/0ea817f020274f678061c27bbd199a874a2578e8) . thank you!